File: src/app/Class/user/UserAccount.ts
export class UserAccount {

    private userName: string;
    private password: string;
    private email: string;

    constructor(obj?: any) {
      this.userName = (obj && obj.userName) || '';
      this.password = (obj && obj.password) || '';
      this.email = (obj && obj.email) || '';
    }

    // Getters

    getUserName(): string {
      return this.userName;
    }

    getPassword(): string {
      return this.password;
    }

    getEmail(): string {
      return this.email;
    }

    // Setters

    setUserName(value: string) {
      this.userName = value;
    }

    setPassword(value: string) {
      this.password = value;
    }

    setEmail(value: string) {
      this.email = value;
    }

}

File: src/main/java/MusicCube/user/UserAccount.java
package MusicCube.user;

import MusicCube.entities.User;

public class UserAccount {

    private int id;
    private String userName;
    private String password;
    private String email;

    // Constructors

    public UserAccount() {
    }

    public UserAccount(User user) {
        this.id = user.getId();
        this.userName = user.getUserName();
        this.password = user.getPassword();
        this.email = user.getEmail();
    }

    public UserAccount(int id, String userName, String password, String email) {
        this.id = id;
        this.userName = userName;
        this.password = password;
        this.email = email;
    }

// Getters and Setters


    public int getId() {
        return id;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

}
